tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodAttributeExecuteIntegration.cs

```diff
@@ -159,6 +162,77 @@ internal static CallTargetReturn<TReturn> OnMethodEnd<TTarget, TReturn>(TTarget
                     returnValue = (TReturn)GetFinalResults(results);
                 }
             }
+            else if (CIVisibility.Settings.FlakyRetryEnabled == true && resultStatus == TestStatus.Fail)
+            {
+                // Flaky retry is enabled and the test failed
+                Interlocked.CompareExchange(ref _totalRetries, CIVisibility.Settings.TotalFlakyRetryCount, -1);
+                var remainingRetries = CIVisibility.Settings.FlakyRetryCount;
+                if (remainingRetries > 0)
+                {
+                    // Handle retries
+                    var results = new List<IList>();
+                    results.Add(returnValueList);
+                    for (var i = 0; i < remainingRetries; i++)
+                    {
+                        if (Interlocked.Decrement(ref _totalRetries) <= 0)
+                        {
+                            Common.Log.Debug<int>("FlakyRetry: Exceeded de number of total retries. [{Number}]", CIVisibility.Settings.TotalFlakyRetryCount);
+                            break;
+                        }
+
+                        Common.Log.Debug<int>("FlakyRetry: [Retry {Num}] Running retry...", i + 1);
+                        var retryTest = MsTestIntegration.OnMethodBegin(testMethod, testMethod.Type, isRetry: true);
```

Comment:
The ~50 lines here seem to be basically a direct copy of the `isTestNew` branch above. Is it worth extracting to a helper method? The only difference I see is in how the results are collected?

Reply:
I did a refactor in [5d5ae1e](https://github.com/DataDog/dd-trace-dotnet/pull/6061/commits/5d5ae1e45875fdfa5193a13eaf0fe80a146bd473)